### ISSUE TITLE
Handle a scenario where a user tries to remove a trackable that's still being added

### ISSUE THREAD

**Issue by KacperKluka** [ably/ably-asset-tracking-swift#342]
There can be a situation in which a trackable is being added and a moment later it is being removed. That's because both `add()` and `remove()` operations are async ones. What happens now is not predictable as the result depends on whether the trackable adding operation was completed before the removal began. What should happen is that the trackable should always be removed if someone called the `remove()` method. So if the trackable that's requested to be removed is currently in the add process we should stop that process, revert all operations that were performed until now and then notify that the removal was successful.

On Android, we have a [`TrackableRemovalGuard`](https://github.com/ably/ably-asset-tracking-android/blob/main/publishing-sdk/src/main/java/com/ably/tracking/publisher/guards/TrackableRemovalGuard.kt) that's used to mark trackables for removal if they're still during the add process. Then each step of the add process checks if the trackable was marked for removal and if it was they revert the add process.

**Comment by lawrence-forooghian** [comment]
This should be done alongside #341.

**Comment by lawrence-forooghian** [comment]
@KacperKluka What should be the result of the in-progress add operation? Should it fail?

**Comment by KacperKluka** [comment]
Yes, the in-progress add operations should fail with a special exception ([here's](https://github.com/ably/ably-asset-tracking-android/blob/main/publishing-sdk/src/main/java/com/ably/tracking/publisher/workerqueue/workers/TrackableRemovalRequestedWorker.kt#L23-L26) how it works on Android).

**Comment by deanna-lad** [comment]
The corresponding Jira issue is https://ably.atlassian.net/browse/SDK-2594
